File: app/models/bdo_item.py
from sqlalchemy import Column, Integer, String, Boolean

from app.database.base_class import Base


class BdoItem(Base):
    __tablename__ = 'bdo_item'

    id = Column(Integer, primary_key=True, autoincrement=True)
    item_num = Column(Integer, unique=True)
    item_name_kr = Column(String(20), index=True)
    max_enhancement_level = Column(Integer)
    grade = Column(Integer)
    category_num = Column(String(8))
    category_name_kr = Column(String(20))
    tradeable = Column(Boolean, nullable=False)

File: src/main/kotlin/kr/karanda/karandaserver/entity/BDOItem.kt
package kr.karanda.karandaserver.entity

import jakarta.persistence.*
import kr.karanda.karandaserver.dto.BDOItem as BDOItemDTO

@Entity(name = "bdo_item")
class BDOItem(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null,
    @Column(unique = true)
    var itemNum: Int,
    var itemNameKr: String,
    var itemNameEn: String?,
    var maxEnhancementLevel: Int,
    var grade: Int,
    var categoryNum: String,
    var categoryNameKr: String,
    @Column(name = "tradeable", nullable = false)
    var tradeAble: Boolean,
) {
    fun toDTO() = BDOItemDTO(
        itemNum = itemNum,
        itemNameKr = itemNameKr,
        maxEnhancementLevel = maxEnhancementLevel,
        grade = grade,
        categoryNum = categoryNum,
        categoryNameKr = categoryNameKr,
        tradeAble = tradeAble,
        itemNameEn = itemNameEn,
    )
}